### ISSUE TITLE
Incorrect narrowing on assignment when assigned value is class with `...` for ParamSpec specialization

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#5757]
### Discussed in https://github.com/microsoft/pyright/discussions/5752

Here's a minimal repro:

```python
from typing import Callable, Generic, TypeVar, ParamSpec

S = TypeVar("S")
P = ParamSpec("P")

class Parent(Generic[S]):
    def __init__(self, s: S):
        ...

class Child(Parent[S], Generic[S, P]):
    def __init__(
        self, s: S, t_callable: Callable[P, int], *args: P.args, **kwargs: P.kwargs
    ):
        ...

def func1(s: str) -> int:
    return int(s)

def func2():
    v1: Parent[int] | Child[int, ...] = Child(1, func1, "1")
    reveal_type(v1, expected_text="Child[int, (s: str)] | Child[int, ...]")
```

**Comment by erictraut** [comment]
Here's an even simpler repro that doesn't involve ParamSpecs.

```python
from typing import Any, Generic, TypeVar

S = TypeVar("S")
T = TypeVar("T")

class Parent(Generic[S]): ...
class Child(Parent[S], Generic[S, T]): ...

v1: Parent[int] = Child[int, Any]()
reveal_type(v1, expected_text="Child[int, Any]")
```

**Comment by erictraut** [comment]
This is included in pyright 1.1.324, which I just published. It will also be included in a future release of pylance.
